### PR TITLE
#800 correct handling parameter value empty array

### LINKED ISSUE #800: empty array parameter value encodes incorrectly
## Basic Info

parameters like `{ foo: { bar: [] } }` and `{ foo: [] }` encoded incorrectly

it's just being skipped. I don't think this is right behavior

related to https://github.com/JsonApiClient/json_api_client/issues/278

* Faraday Version:
0.15.1 and master github branch
* Ruby Version:
2.2.4

## Steps to reproduce
If possible, please provide the steps to reproduce the issue.
```ruby
>> Faraday::NestedParamsEncoder.encode(foo: { bar: [] }, baz: [])
=> "&"
```

I think correct behavior will be to encode it to `"baz%5B%5D&foo%5Bbar%5D%5B%5D"` or `"baz%5B%5D=&foo%5Bbar%5D%5B%5D="` or `"baz&foo%5Bbar%5D"`, `"baz=&foo%5Bbar%5D="`

### PR DESCRIPTION
**Pull request by senid231**
## Description
correct handling of parameters when value is an empty array
fix #800

## Todos
List any remaining work that needs to be done, i.e:
- [x] Tests
- [ ] Documentation *(not sure where it should be documented)*

## Additional Notes
Optional section
will fix https://github.com/JsonApiClient/json_api_client/issues/278

### CODE REVIEW

File: lib/faraday/parameters.rb [hunk 1]
```diff
@@ -39,13 +39,15 @@ def self.encode(params)
...
           return buffer.chop
-        elsif value.is_a?(Array)
+        elsif value.is_a?(Array) && value.size != 0
           buffer = ""
           value.each_with_index do |val, i|
             new_parent = "#{parent}%5B%5D"
             buffer << "#{to_query.call(new_parent, val)}&"
           end
           return buffer.chop
+        elsif value.is_a?(Array) && value.size == 0
```
Comment (olleolleolle) [inline lib/faraday/parameters.rb]: Perhaps the `Array#empty?` method is even clearer to read?

Comment (iMacTia) [reply lib/faraday/parameters.rb]: I'd go a step further and merge this 2 branches:

```ruby
elsif value.is_a?(Array)
  return "#{parent}%5B%5D" if value.empty? # This is the only new line
  buffer = ""
  value.each_with_index do |val, i|
    new_parent = "#{parent}%5B%5D"
     buffer << "#{to_query.call(new_parent, val)}&"
  end
  return buffer.chop
```

Comment (senid231) [reply lib/faraday/parameters.rb]: nice catch
fixed

**Comment by iMacTia** [comment]
Nice optimisation, looks good now.

**Pull request merged**